controllers/reconcile.go

```diff
@@ -216,30 +216,58 @@ func makeAkamaiEdgercFileFromSecret(secret *v1.Secret) string {
 }
 
 func createCachePurgePathList(frontend *crd.Frontend, frontendEnvironment *crd.FrontendEnvironment) []string {
-	// Set purgeHost by ensuring the URL begins with https:// and has no trailing /
-	purgeHost := strings.TrimSuffix(fmt.Sprintf("https://%s", strings.TrimPrefix(frontendEnvironment.Spec.AkamaiCacheBustURL, "https://")), "/")
+	var purgePaths []string
 
-	// Initialize with a default path if AkamaiCacheBustPaths is nil
-	purgePaths := []string{fmt.Sprintf("%s/apps/%s/fed-mods.json", purgeHost, frontend.Name)}
+	// Helper function to check if a path is already in the list
+	contains := func(slice []string, item string) bool {
+		for _, existing := range slice {
+			if existing == item {
+				return true
+			}
+		}
+		return false
+	}
 
-	if frontend.Spec.AkamaiCacheBustPaths == nil {
+	// Return early if there are no AkamaiCacheBustURLs
+	if frontendEnvironment.Spec.AkamaiCacheBustURLs == nil {
 		return purgePaths
 	}
 
-	purgePaths = make([]string, 0, len(frontend.Spec.AkamaiCacheBustPaths))
-	for _, path := range frontend.Spec.AkamaiCacheBustPaths {
-		// Check if path is a full URL (starts with "http://" or "https://")
-		if strings.HasPrefix(path, "http://") || strings.HasPrefix(path, "https://") {
-			// Add full URL path directly
-			purgePaths = append(purgePaths, path)
-		} else {
-			// Ensure each path has a leading slash but no double slashes
-			if !strings.HasPrefix(path, "/") {
-				path = "/" + path
+	for _, cacheBustURL := range frontendEnvironment.Spec.AkamaiCacheBustURLs {
+		// Ensure the URL begins with https:// and has no trailing /
+		purgeHost := strings.TrimSuffix(fmt.Sprintf("https://%s", strings.TrimPrefix(cacheBustURL, "https://")), "/")
```

Comment:
I think I'm getting confused here with the `strings.TrimPrefix()` portion of this. It looks like the prefix `https://` gets trimmed off of the string variable `cacheBustURL` only to have it added back through the string formatting of `fmt.Sprintf("https://%s")`. This is probably just confusing me.

Reply:
We discussed this in chat, but this line ensures that regardless of whether or not the provided URL starts with `"https://"` it ends up with it. There may be a better, more idiomatic way to do this, this was just want I landed on in an effort to avoid introducing a conditional.